tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,212 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Virtualization Extend LVM Storage pool / volume test
+#
+# Test flow:
+# - Check a additional free hard disk
+# - Wipe hard disk clean
+# - Partition test disk
+# - Create a pv and display result
+# - Create a vg named 'lvm_vg' and display result
+# - Create a storage pool (lv) named 'guest_image_lvm'
+# - Create a logical volume (lv) size 1G
+# - Attach a logical volume (lv) to guest systems
+# - Clone a logical volume (lv)
+# - Cleanup
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use virt_autotest::utils;
+use virt_autotest::common;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_utils;
+use version_utils 'is_sle';
+
+our $lvm_vg_name = 'lvm_vg';
+our $lvm_pool_name = 'guest_image_lvm';
+our $timeout = 180;
+
+sub run_test {
+    my ($self) = @_;
+
+    record_info "Prepare Guest Systems";
+    foreach ( keys %virt_autotest::common::guests ) {
+        start_guests() unless is_guest_online($_);
+    }
+
+    ## About LVM volumes management
+    my ( $sys_disk, $virt_disk, $dev, $lvm_disk );
+
+    # Check with all existed Hard disks
+    my @disks = split( /\n/, script_output("lsblk -n -l -o NAME -d -e 7,11") );
+    my $scalar = @disks;
+
+    # Requires at least 2 physical hard disks for LVM Storage test
+    if ( ( $scalar eq 1 ) || get_var('KEEP_DISKS') ) {
+        record_info(
+            "WARNING",
+            "Requires at least 2 physical hard disks for LVM Storage test\n",
+            result => 'softfail'
+        );
+        return;
+    }
+
+    # Find out the used(mounted) hard disks
+    $sys_disk = script_output 'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /)';
+    record_info "System Disk:", "$sys_disk";
+    record_info( "List Block Devices", script_output("lsblk") );
+
+    # Use a additional free hard disk for LVM volumes
+    $dev = "/dev/";
+    foreach my $disk (@disks) {
+        if ( $disk ne $sys_disk ) {
+            if ( is_sle '15+' ) {
+                $virt_disk = script_output
+                  'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /var/lib/libvirt)';
+                record_info "Virt Disk:", "$virt_disk";
+                if ( $disk ne $virt_disk ) {
+                    $lvm_disk = $dev . $disk;
+                    last;
+                }
+            }
+            else {
+                $lvm_disk = $dev . $disk;
+                last;
+            }
+        }
+    }
```

Comment:
Honestly speaking, the above algorithm does not guarantee the found disk is free.

Reply:
I think there is way to find free or unused disk on linux host:
```
Find all disks
host:~ # lsblk --list -o NAME,TYPE,STATE | grep -i disk | grep running
sda  disk running
sdb  disk running

Find all mountpoints
host:~ # lsblk --list -o NAME,MOUNTPOINTS
NAME MOUNTPOINTS
sda  
sda1 
sda2 
sda3 
sdb  
sdb1 
sdb2 /usr/local
     /srv
     /opt
     /home
     /boot/writable
     /boot/grub2/x86_64-efi
     /boot/grub2/i386-pc
     /.snapshots
     /root
     /
sdb3 /var
sr0  

Find free or unused disk. For any disk in "All Disks", if it has no point in "ALL MOUNTPOINTS", it is free or unused disk
host:~ # lsblk --list -o NAME,MOUNTPOINTS | grep -i sda | awk '{print $2}'




host:~ # lsblk --list -o NAME,MOUNTPOINTS | grep -i sdb | awk '{print $2}'


/usr/local
/var

So sda is free disk in this case
```

Reply:
> Honestly speaking, the above algorithm does not guarantee the found disk is free. It just guarantees it is not system and virt disk.

from my side, if It just guarantees it is not system and virt disk. it should be unused disk on our SLE KVM and XEN vm host.

Reply:
If you leave a comment saying "Find free or unused disk", then current code does not look good enough.

Reply:
yes, update code to find out unused hard disk.

Reply:
Leave this change to be confirmed by @waynechen55 .